packages/hash/design-system/src/use-scroll-lock.ts

```diff
@@ -3,54 +3,59 @@ import { useEffect, useLayoutEffect, useRef, useState } from "react";
 /**
  * @see https://github.com/mui/material-ui/blob/master/packages/mui-utils/src/getScrollbarSize.ts
  */
-const getScrollbarSize = (doc: Document) => {
-  const documentWidth = doc.documentElement.clientWidth;
+const getScrollbarSizeOfDocument = () => {
+  const documentWidth = document.documentElement.clientWidth;
   return Math.abs(window.innerWidth - documentWidth);
 };
 
-const useLastScrollbarSize = () => {
+const useLastScrollbarSize = (element?: HTMLElement) => {
   const [lastScrollbarSize, setLastScrollbarSize] = useState(0);
   const observerRef = useRef<ResizeObserver>();
 
   useEffect(() => {
     observerRef.current = new ResizeObserver(() => {
-      const scrollbarSize = getScrollbarSize(document);
+      const scrollbarSize = element
+        ? element.offsetWidth - element.clientWidth
+        : getScrollbarSizeOfDocument();
 
       if (scrollbarSize > 0) {
         setLastScrollbarSize(scrollbarSize);
       }
     });
 
-    observerRef.current.observe(document.body);
+    observerRef.current.observe(element ?? document.body);
```

Comment:
You've put `document.body` as the default in multiple places. You can keep `element` as required in `useLastScrollbarSize` and just pass it from `useScrollLock`

Reply:
Updated here https://github.com/hashintel/hash/pull/1751/commits/c0ac496c437d3eef0b0d97f21492359d945f0744